### PR TITLE
Loki: Add more assertions to loki-query-builder.spec.ts

### PR DESCRIPTION
**Pull request by itsgareth**
**What this PR does / why we need it**:
This adds more assertions to the loki query builder e2e tests.
Tested the `Raw query` toggle.
Tested the `Explain query` toggle.
Checks the query is visible when changing to the `Code` editor.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: e2e/various-suite/loki-query-builder.spec.ts [hunk 1]
```diff
@@ -14,6 +14,16 @@ const addDataSource = () => {
   });
 };
 
+// We need this because the final query is split into separate DOM elements using e2e().contains(finalQuery).should('not.exist'); does not detect the query.
+const checkFinalQuery = (query: 'be.visible' | 'not.exist') => {
+  const keywords = ['rate', 'instance1|instance2', 'logfmt', '__error__', '$__interval'];
+  for (const word of keywords) {
+    e2e().contains(word).should(query);
+  }
+};
```
Comment (matyax) [inline e2e/various-suite/loki-query-builder.spec.ts:23]: Looks like `checkFinalQuery` is just called once, and with `'be.visible'`?

```suggestion
const checkFinalQuery = () => {
  const keywords = ['rate', 'instance1|instance2', 'logfmt', '__error__', '$__interval'];
  for (const word of keywords) {
    e2e().contains(word).should('be.visible');
  }
};
```

@ivanahuckova

**Review by matyax — APPROVED** [review]
Good work with e2e testing! Left some questions and suggestions.

**Review by matyax — CHANGES_REQUESTED** [review]
I meant to just `comment`, not `approve`.

File: e2e/various-suite/loki-query-builder.spec.ts (hunk 1 above)
Comment (ivanahuckova) [reply e2e/various-suite/loki-query-builder.spec.ts:23]: This is @gwdawson 's PR, so I am going to let him get back to you on this. 🙂

Comment (matyax) [reply e2e/various-suite/loki-query-builder.spec.ts:23]: Looking forward to it 🧐

Comment (matyax) [reply e2e/various-suite/loki-query-builder.spec.ts:23]: Looking forward to it 🧐. By the way, I just realized that my mention was aimed at the other comment, not this one.

Comment (gwdawson) [reply e2e/various-suite/loki-query-builder.spec.ts:23]: this was initially used for both `be.visible` and `not.exist` but i guess i ended up replacing the second function call and didn't update the function.

File: e2e/various-suite/loki-query-builder.spec.ts [hunk 2]
```diff
@@ -72,13 +80,16 @@ describe('Loki query builder', () => {
...
-    for (const word of finalQuery.split(' ')) {
-      e2e().contains(word).should('be.visible');
-    }
+    // Toggle raw query
+    e2e().contains('label', 'Raw query').click();
+    e2e().contains('Raw query').should('have.length', 1);
+
+    // Change to code editor
+    e2e().contains('label', 'Code').click();
+    checkFinalQuery('be.visible');
```
[the remark this answers is not in the document]
Comment (gwdawson) [reply e2e/various-suite/loki-query-builder.spec.ts:89]: @matyax you raise a good point here about `reading the tests like a book`. now the helper function is only being used once it makes more sense to just to list the tests like your suggested change, i'll update this now.

[the remark this answers is not in the document]
Comment (matyax) [reply e2e/various-suite/loki-query-builder.spec.ts:89]: Fantastic, thank you!

**Review by matyax — APPROVED** [review]
👌

**Pull request merged**